src/patterns/gen-ai/aws-llama-index-data-loader/index.ts

```diff
@@ -0,0 +1,493 @@
+  private readonly dockerImageAssetDirectory: string;
+  private readonly memoryLimitMiB: number;
+  private readonly containerLoggingLevel: string;
+
+  constructor(scope: Construct, id: string, props: LlamaIndexDataLoaderProps) {
+    super(scope, id);
+
+    const baseProps: BaseClassProps = {
+      stage: props.stage,
+      constructName: ConstructName.AWSLLAMAINDEXDATALOADER,
+      constructId: id,
+      observability: props.observability,
+    };
+
+    this.updateEnvSuffix(baseProps);
+    this.addObservabilityToConstruct(baseProps);
+
+    // Update the optional properties to their defaults
+    this.dockerImageAssetDirectory = props.dockerImageAssetDirectory ?? join(
+      __dirname, '..', '..', '..', '..', 'resources', 'gen-ai', 'aws-llama-index-data-loader', 'docker',
+    );
+    this.memoryLimitMiB = props.memoryLimitMiB ?? 2048;
+    this.containerLoggingLevel = props.containerLoggingLevel ?? 'WARNING';
+    let bucketsInvolved = [];
+    if (props.outputBucket) {
+      this.outputBucket = props.outputBucket;
+    } else {
+      this.bucketKey = new Key(this, 'LogBucketKey', {
+        enableKeyRotation: true,
+      });
+      this.logBucket = new Bucket(this, 'LogBucket', {
+        enforceSSL: true,
+        versioned: true,
+        encryption: BucketEncryption.KMS,
+        encryptionKey: this.bucketKey,
+        bucketKeyEnabled: true,
+        removalPolicy: RemovalPolicy.DESTROY,
+        autoDeleteObjects: true,
+        objectLockEnabled: true,
+        objectLockDefaultRetention: {
+          mode: ObjectLockMode.GOVERNANCE,
+          duration: Duration.days(1),
+        },
+        blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+      });
+      bucketsInvolved.push(this.logBucket);
+      this.outputBucket = new Bucket(this, 'Output', {
+        enforceSSL: true,
+        versioned: true,
+        serverAccessLogsBucket: this.logBucket,
+        serverAccessLogsPrefix: 'output-bucket-access-logs',
+        encryption: BucketEncryption.KMS,
+        encryptionKey: this.bucketKey,
+        bucketKeyEnabled: true,
+        blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+        removalPolicy: RemovalPolicy.DESTROY,
+        autoDeleteObjects: true,
+        objectLockEnabled: true,
+        objectLockDefaultRetention: {
+          mode: ObjectLockMode.GOVERNANCE,
+          duration: Duration.days(1),
+        },
+        lifecycleRules: [
+          {
+            id: 'AbortIncompleteMultipartUpload',
+            enabled: true,
+            abortIncompleteMultipartUploadAfter: Duration.days(1),
+          },
+        ],
+      });
+    }
+    bucketsInvolved.push(this.outputBucket);
+    // Create a new SSM Parameter holding a String
+    const circuitBreakerParameter = new StringParameter(this, 'CircuitBreaker', {
+      stringValue: 'False',
+    });
+    const asset = new DockerImageAsset(this, 'Image', {
+      directory: this.dockerImageAssetDirectory,
+      platform: Platform.LINUX_AMD64,
```

Comment:
why are we setting this to Linux AMD 64 ?

Reply:
Due to macOS, the base image doesn't have a tag for the platform like the lambda does, so defining that explicitely so alternative libraries that are pulled in from Docker's build do not mix in for python. This version has fixed to x86_64 to reduce issues such as https://github.com/aws/aws-cdk/issues/30962